db/migrate/20201217163340_update_enterprise_socia_links.rb

```diff
@@ -1,14 +1,11 @@
 class UpdateEnterpriseSociaLinks < ActiveRecord::Migration
-  def instagram_regex
-    %r{\A(?:https?://)?(?:www\.)?instagram\.com/([a-zA-Z0-9._@-]{1,30})/?\z}
-  end
   def change
     say_with_time "Updating enterprises..." do
       count = 0
       Enterprise.find_each do |enterprise|
         unless enterprise.instagram.nil?
           say "Update enterprise: #{enterprise.id}"
-          enterprise.update(instagram: enterprise.instagram.downcase.try(:gsub, instagram_regex, '\1').gsub('@', ''))
+          enterprise.update(instagram: enterprise.instagram.downcase.try(:gsub, enterprise.instagram_regex, '\1').delete('@'))
```

Comment:
If you copy the Enterprise model with the `instagram=` method you wrote, you can simplify this to:
```suggestion
          enterprise.update(instagram: enterprise.instagram)
```

The `update` method will call `instagram=` which you already implemented.